Add option to limit the number of lines used for guessing in io.ascii (#16840)
astropy/astropy: docs/whatsnew/7.0.rst

Note that it is not yet possible to use ``Quantity`` with dask arrays directly.

.. _whatsnew_7_0_faster_io_ascii_guessing:

Faster guessing of formats in ``astropy.io.ascii``
==================================================

When reading text-based tables with ``astropy.io.ascii`` and without specifying the
table format, guessing the format should now be faster for large tables as it is now

> based on a subset of the table - by default the first 10000 lines of the file. If

Reviewer: I think some slight variation of my suggestion for the change log would work here to make a more direct statement about the performance improvement.

Author: Done

needed, this limit can be modified with the following configuration item:

.. code-block:: python

    >>> from astropy.io.ascii import conf
    >>> conf.guess_limit_lines = 100000

> and the limit can be disabled by setting this value to `None`. For large tables, it
> is however preferable to specify the format of the table explicitly when reading.

Reviewer: ```suggestion
and the limit can be disabled by setting this value to `None`. When reading large tables
you should specify the format of the table explicitly if possible.
```
